### PR TITLE
basic functionality

### PR DESCRIPTION
**Pull request by AnitaTudor**
This PR  add the basic functionality of reservation REST API: 
- "/reservations" endpoint is for getting all the reservation stored in the database
- "/checkReservations/{identifier}" endpoint is for checking the reservation referenced by the Identifier(A random generated alphabetical string). If the reservation is not found a message is shown.
- "/deteleReservation/{identifier}" endpoint is for deleting the reservation .

### CODE REVIEW

File: src/main/java/ro/unibuc/hello/exception/ReservationNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package ro.unibuc.hello.exception;
+
+public class ReservationNotFoundException extends RuntimeException{
+    public ReservationNotFoundException(String identifier) {
+        super("Reservation " + identifier+" does not seem to exist!");
```
Comment (vladpirlog) [inline src/main/java/ro/unibuc/hello/exception/ReservationNotFoundException.java:5]: nit: leave a blank after `RuntimeException` and `identifier` (e.g.: `identifier + " does..."`)

File: src/main/java/ro/unibuc/hello/repository/ReservationRepository.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * No need to implement this interface.
+ * Spring Data MongoDB automatically creates a class it implementing the interface when you run the application.
+ */
+@Repository
+public interface ReservationRepository extends MongoRepository<Reservation, String> {
+    Optional<Reservation> findByIdentifier(String identifier);
+    Optional<Reservation> deleteByIdentifier(String identifier);
+
+
+}
```
Comment (vladpirlog) [inline src/main/java/ro/unibuc/hello/repository/ReservationRepository.java:20]: nit: too many empty lines

File: src/main/java/ro/unibuc/hello/HelloApplication.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+@SpringBootApplication
+public class HelloApplication {
+
+	@Autowired
+	private ReservationRepository reservationRepository;
+	@Autowired
+	private GenerateIdentifier generateIdentifier;
+
+
```
Comment (vladpirlog) [inline src/main/java/ro/unibuc/hello/HelloApplication.java:22]: nit: too many empty lines

File: src/main/java/ro/unibuc/hello/controller/ReservationController.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    private ReservationService reservationService;
+
+    private final AtomicLong counter = new AtomicLong();
+
+    @GetMapping("/reservations")
+    @ResponseBody
+    public List<Reservation> showReservations() {
+        return reservationService.getAllReservations();
+    }
+    @GetMapping("/checkReservation/{identifier}")
```
Comment (vladpirlog) [inline src/main/java/ro/unibuc/hello/controller/ReservationController.java:29]: nit: empty line between methods

File: src/main/java/ro/unibuc/hello/utils/GenerateIdentifier.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            StringBuilder sb = new StringBuilder();
+
+            Random random = new Random();
+
+
+            int length = 7;
+
+                for (int i = 0; i < length; i++) {
+
+                int index = random.nextInt(alphabet.length());
```
Comment (vladpirlog) [inline src/main/java/ro/unibuc/hello/utils/GenerateIdentifier.java:23]: nit: the `for` statement is indented too much

**Review by vladpirlog — CHANGES_REQUESTED** [review]

File: src/main/java/com/javafrenzy/restaurant/controller/ReservationController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+
+@Controller
+public class ReservationController {
+
+    @Autowired
+    private ReservationService reservationService;
+
+    private final AtomicLong counter = new AtomicLong();
+
```
Comment (vladpirlog) [inline src/main/java/com/javafrenzy/restaurant/controller/ReservationController.java:23]: is this used anywhere?

**Review by vladpirlog — CHANGES_REQUESTED** [review]

**Review by AnitaTudor — COMMENTED** [review]
changed files according to vlad feedback

**Review by vladpirlog — APPROVED** [review]

**Pull request merged**